modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java

```diff
@@ -282,6 +292,10 @@ private LockState lockState(LockKey key) {
                 if (v == null) {
                     res[0] = slots[index];
                     assert !res[0].markedForRemove;
+
+                    if (isSlotsExhausted.compareAndExchange(false, true)) {
```

Comment:
Where do you reset isSlotsExhausted back to false?

Reply:
Only the first time we are logging it.

Reply:
What do you mean? On first logging it's updated from false to true. I'm asking about resetting back from true to false.

Reply:
I made the output through the interval.